python/ray/scripts/scripts.py

```diff
@@ -604,8 +604,7 @@ def start(node_ip_address, address, port, redis_password, redis_shard_ports,
             if reachable:
                 ray_params.update_if_absent(
                     external_addresses=external_addresses)
-                if len(external_addresses) > 1:
-                    num_redis_shards = len(external_addresses) - 1
+                num_redis_shards = len(external_addresses)
```

Comment:
Previously we use the second to the last external Redis instance as shards, and now we are changing it to use all external Redis instances as shards? Is there any reason to do so?

Reply:
External redis is little bit different from internal one.
When use internal redis, we start primary instance in the head node and sharding instances per `shard ports` specified, which means we always have sharding.
But with external redis, we allow only one remote redis provided, that means we must reuse the primary as a shard. We can always reuses it in external redis supporting. Even it is little different from internal one, at least it is consistent with external no matter the number of redis instances is 1 or more.

Reply:
What I expect is, when there is only 1 external Redis instance provided, we use it for both primary and sharding, and when there is more than 1 external Redis instance provided, we use the first one for primary and the others for sharding. Does that sound good to you?

Reply:
To be honest that's my initial idea too. The reason why I changed it is that I think always reusing primary instance as shard is a natural idea. Telling difference between 1 or more remote redis instances will make codes complicated and not easy to explain in document.
We didn't reuse primary instance in local because we can start as many redis server in local as we want. I think that's the main difference between local and remote.
My preference is not very strong, I just want find a better one. What do you think?